examples/platform/esp32/time/TimeSync.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "TimeSync.h"
+#include <esp_sntp.h>
+#include <lib/support/CHIPMemString.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+#define REF_TIME 1546300800 /* 01-Jan-2019 00:00:00 */
+/* Timer interval once every day (24 Hours) */
+#define TIME_PERIOD_SEC 86400000ULL
+namespace {
+const uint8_t kMaxNtpServerStringSize = 128;
+char mSntpServerName[kMaxNtpServerStringSize];
+
+static CHIP_ERROR GetLocalTimeString(char * buf, size_t buf_len)
+{
+    struct tm timeinfo;
+    char strftime_buf[64];
+    time_t now;
+    time(&now);
+    localtime_r(&now, &timeinfo);
+    if (strftime(strftime_buf, sizeof(strftime_buf), "%Y-%m-%dT%H:%M:%S%z", &timeinfo) == 0)
+    {
+        ChipLogError(DeviceLayer, "Buffer too small");
+        return CHIP_ERROR_BUFFER_TOO_SMALL;
+    }
+    size_t print_size = snprintf(buf, buf_len, "%s, DST: %s", strftime_buf, timeinfo.tm_isdst ? "Yes" : "No");
+    if (print_size >= buf_len)
```

Comment:
`The number of characters that would have been written if n had been sufficiently large, not counting the terminating null character.`

This should be:
if (print_size >= (buf_len - 1))

Reply:
Updated. Thanks